src/js/main.js

```diff
@@ -50,64 +58,65 @@ if (navigator && navigator.clipboard && navigator.permissions) {
 	});
 }
 
-// When someone navigates directly to a checklist item using its "Share Link" 
-// like a11yproject.com/checklist/#validate-your-html, the item with the
-// matching id attribute will be scrolled into view. Then, if JS is enabled,
-// this code will open its associated <details> element. 
-function openLinkedCheckListItem() {			
-	var checklistItems = document.querySelectorAll("[data-checklist-item-id]");
-
-	checklistItems.forEach(function (item) {
-		var shareLink = "#" + item.getAttribute("data-checklist-item-id");
-		var isItemLinked = shareLink === document.location.hash;
-
-		if (isItemLinked) {
-			item.setAttribute("open", true);
-		}
-	});
+/**
+ * If someone has opened the `checklist` page using one a checklist "Share link"
+ * (ex: a11yproject.com/checklist/#validate-your-html) the item with the
+ * corresponding id will scroll into view. Then, if JS is enabled,
+ * this function will open its associated <details> element.
+ */
+function openLinkedCheckListItem() {
+	var hash = window.location.hash.substr(1);
+	var checklistItem = document.querySelector(
+		"[data-checklist-item-id=" + hash + "]"
+	);
+
```

Comment:
This is great improvement! Only in the comment section, I would change it to `If someone opens the `checklist` page using a checklist "Share link" (ex: a11yproject.com/checklist/#validate-your-html) the item with the corresponding id will scroll into view. Then, if JS is enabled, this function will open its associated <details> element.`